### DISCUSSION TITLE
Zsh auto suggest plugins don't work

### DISCUSSION THREAD

**Discussion by RomanAverin** [ghostty-org/ghostty#4021] [Feature Requests, Ideas] [answered]
Strange behavior. Some plugins don't work, for example zsh-autosuggestions. In other emulators it works
The shell integration option is available in the configuration.
`shell-integration = zsh`

**Comment by 00-kat** [comment]
Do you have any other information?  What's your Ghostty configuration?  Does `shell-integration = none` solve it?

**Reply by RomanAverin** [reply]
My apologies, it seems to be an issue with the Flatland color scheme

**Reply by Quba1** [reply]
Same issue with `OneHalfDark` color scheme

**Reply by HadyMash** [reply]
Same here with `OneHalfDark`, did you find any fix?

**Comment by pchalasani** [comment]
I had the same issue with Builtin Solarized Dark, i.e. I couldn't see the zsh-autosuggestions. However rather than change the theme, I want to stick with it, and somehow customize the autosuggestion colors, like I was able to do with iterm2 (I think it was "faint text opacity", I can't recall). Any ideas how I can do that? Thanks

**Reply by HadyMash** [reply]
Have you found a way to do that?

**Reply by Quba1** [reply, marked as answer]
If you want to keep the selected theme (because changing it is probably the easiest option), zsh-autosuggestions provide a config option `ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE.

From the [docs](https://github.com/zsh-users/zsh-autosuggestions/tree/master?tab=readme-ov-file#suggestion-highlight-style):

> Set ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE to configure the style that the suggestion is shown with. The default is fg=8, which will set the foreground color to color 8 from the [256-color palette](https://upload.wikimedia.org/wikipedia/commons/1/15/Xterm_256color_chart.svg).
>...
>Background color can also be set, and the suggestion can be styled bold, underlined, or standout. For example, this would show suggestions with bold, underlined, pink text on a cyan background:

```bash
ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE="fg=#ff00ff,bg=cyan,bold,underline"
```

I haven't tried this configuration personally, but seems like a viable solution.

@HadyMash

**Reply by HadyMash** [reply]
Ok thanks i'll give it a try. I don't wanna change the theme cus im a one dark pro fanatic 😅. I'll let you know if it works or not. thanks!

**Reply by HadyMash** [reply]
Thanks @Quba1 that worked.

In case anyone else comes across this, I'm using the OneHalfDark colour scheme so i used `#818386` as the fg and didn't need to set anything else.

in my `.zshrc`:
```zsh
# Ghostty integration
if [[ -n $GHOSTTY_RESOURCES_DIR ]]; then
  # change zsh autosuggestion color so it's not invisible
  ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE='fg=#818386'
  source "$GHOSTTY_RESOURCES_DIR"/shell-integration/zsh/ghostty-integration
fi
```

I've put it inside the [shell integration](https://ghostty.org/docs/features/shell-integration) if condition to not affect other terminals.

The above produces the following result:

![CleanShot 2025-02-03 at 7  09 46](https://github.com/user-attachments/assets/677b22a4-1064-4381-b624-5cf3fd0105fb)
